File: baekjoon/bj10828.cpp
/*
문제:백준10828번
출처:https://www.acmicpc.net/problem/10828
*/
#include<iostream>
#include<stack>
#include<string>
using namespace std;

int main() {
	int N;
	string c; int num;
	cin >> N;
	stack<int> s;
	while (N--) {
		cin >> c;
		if (c == "push") {
			cin >> num;
			s.push(num);
		}
		else if (c == "pop") {
			if (!s.empty()) {
				cout << s.top() << '\n';
				s.pop();
			}
			else cout << -1 << '\n';
		}
		else if (c == "size") {
			cout << s.size() << '\n';
		}
		else if (c == "empty") {
			if (s.empty())cout << 1 << '\n';
			else cout << 0 << '\n';
		}
		else {
			if (!s.empty()) cout << s.top() << '\n';
			else cout << -1 << '\n';
		}
	}
}

File: baekjoon/bj10828.java
import java.io.*;
import java.util.Stack;
import java.util.StringTokenizer;

public class bj10828 {
    static Stack<Integer> st = new Stack<Integer>();
    static int num,N;
    static String command;

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());

        for(int i=0;i<N;i++){
            StringTokenizer stk = new StringTokenizer(br.readLine());
            command = stk.nextToken();

            if(command.equals("push")){
                num=Integer.parseInt(stk.nextToken());
                st.push(num);
            }else if(command.equals("pop")){
                if(st.isEmpty())
                    System.out.println(-1);
                else System.out.println(st.pop());
            }else if(command.equals("size")){
                System.out.println(st.size());
            }else if(command.equals("empty")){
                if(st.isEmpty())
                    System.out.println(1);
                else System.out.println(0);
            }else if(command.equals("top")){
                if(st.isEmpty())
                    System.out.println(-1);
                else System.out.println(st.peek());
            }
        }

        br.close();
    }
}
